klog.go

```diff
@@ -414,7 +415,12 @@ func init() {
 	logging.addDirHeader = false
 	logging.skipLogHeaders = false
 	logging.oneOutput = false
-	go logging.flushDaemon()
+	go logging.flushDaemon(stop, done)
+	go func() {
+		time.Sleep(20 * time.Second)
+		close(stop)
```

Comment:
This really seems like it will cause the logger to stop logging after 20 seconds (or so, when the stop channel is closed).

I'm wondering if what is really needed is a signal handler, but I worry a bit that adding a signal handler directly here would potentially lead to logging information being dropped too early. 

I kind of wish that klog require explicit initialization rather than being done in init(), since then we could have the caller provide the stop channel and do the signal handling.

I'm not quite sure what the best path forward is from here, not sure if anyone else has some suggestions.

Reply:
Thanks @detiber , you got it right but I am sure that the piece of code before my change was not even working, the init() will exit before it get chance to flush the logs. There is no way the init() function is waiting for go routine flushing the logs. In my case the flush function will be called 4 times. Now question arise, that do we need to make a function which will work other than init().